### PR TITLE
Updating the new hg repo for jdk 10

### PR DESCRIPTION
**Pull request by neomatrix369**
Changing the current source of changesets for JDK 10 from jdk10/jdk10 (http://hg.openjdk.java.net/jdk10/jdk10) to jdk/jdk (http://hg.openjdk.java.net/jdk/jdk/)

As per @Shura, the new announcement points to this change - http://mail.openjdk.java.net/pipermail/jdk-dev/2017-December/000388.html

Will need to change https://ci.adoptopenjdk.net/view/git-hg/job/git-diff-10/ to pass in the extra parameters to the new repo `jdk/jdk`

After merging we run Jenkins job: https://ci.adoptopenjdk.net/view/git-hg/job/git-hg-10/ followed by https://ci.adoptopenjdk.net/view/git-hg/job/git-diff-10/

Scripts impacted:
- /openjdk-build/git-hg/update.sh <=== might not be impacted, we will see once this PR is merged
- ~/openjdk-build/git-hg/diff.sh~

*Usage:*
```
$ diff.sh jdk10 jdk jdk
```
_Old usages will continue to work._

### CODE REVIEW

File: git-hg/diff.sh [hunk 1]
```diff
@@ -15,13 +15,18 @@
 
 set -euo pipefail
 
-version=$1
+git_repo_version=$1
+hg_root_forest=${2:-${1}}                  # for backwards compatibility
+hg_repo_version=${3:-${hg_root_forest_version}}    # for backwards compatibility
```
Comment (karianna) [inline git-hg/diff.sh]: actually where does hg_root_forest_version come from?

Comment (neomatrix369) [reply git-hg/diff.sh]: Typo! changing now!

Comment (neomatrix369) [reply git-hg/diff.sh]: Done!

**Comment by karianna** [comment]
@neomatrix369 Were you able to test this on a jenkins job?  If so then I'm happy to merge

File: git-hg/setup10.sh [hunk 2]
```diff
@@ -21,4 +21,4 @@ git init
 git checkout -b root-commit || exit 1
 git remote add github git@github.com:AdoptOpenJDK/openjdk-jdk10.git
 cd - || exit 1
-bash add-branch.sh jdk10/jdk10
+bash add-branch.sh jdk/jdk
```
Comment (gdams) [inline git-hg/setup10.sh]: Why has this changed to `jdk` instead of `jdk10`?

Comment (neomatrix369) [reply git-hg/setup10.sh]: @Shura has shared this with us:
> As per @shura, the new announcement points to this change - http://mail.openjdk.java.net/pipermail/jdk-dev/2017-December/000388.html

Comment (gdams) [reply git-hg/setup10.sh]: right okay

**Comment by neomatrix369** [comment]
@karianna 
Please don't merge, I haven't tested and wasn't planning to, till we had someone like @gdams or infra team stand-by, just in case the git repo went *ka-boom*! We don't know what will have if we ran the updated scripts on the existing commits - in theory, everyone will say it's going to be fine.

All depends and jenkins jobs depending on them would be affected.

We also need to test setup, update and diff, so we do need a fresh repo to experiment this on. Make a backup of this one, and have an empty one for the new `jdk/jdk` (supposedly jdk10).

I hope you follow my rationale for not doing anything yet.

@gdams are you available next week so we can do this around the same time, we will also need to announce across the channels about it.

**Comment by gdams** [comment]
okay so I am happy for this to be merged, I shall merge this and keep and eye on the job over the weekend.

**Comment by neomatrix369** [comment]
@gdams thanks for doing this, although I have noticed, the `update.sh` script isn't doing what we want Reasons being, the folder `jdk10` on all the agents need to be changed to `jdk` , according the script it will otherwise only download `jdk10` changes.

**Pull request merged**